astropy/io/fits/fitsrec.py

```diff
@@ -1284,6 +1284,19 @@ def _scale_back_ascii(self, col_idx, input_field, output_field):
         if 'D' in format:
             output_field[:] = output_field.replace(b'E', b'D')
 
+    def tolist(self):
+        # In some cases, at least when some columns are VLF, np.recarray.tolist (i.e. of the base class)
+        # can not interpret what is stored in data as list, and instead silently returns undersirable result.
+        # For example in the case of VLF column it an array with variable row length and offsets.
+
+        column_lists = []
+
+        for column in self.columns:
+            column_list = column.array.tolist()
+            column_lists.append(column_list)
```

Comment:
Could be shorter with a list comprehension:
```
column_lists = [column.array.tolist() for column in self.columns]
```

Reply:
Ok, that's how I did it at first but then unfolded it a way which some people might find more readable. No problem to keep it as comprehension.